terraform/modules/region/cloudwatch.tf

```diff
@@ -16,6 +16,126 @@ resource "aws_kms_alias" "cloudwatch_standard_alias" {
   target_key_id = aws_kms_key.cloudwatch_standard.key_id
 }
 
+resource "aws_cloudwatch_log_metric_filter" "uid_service_400_errors" {
+  name           = "${local.environment_name}-uid-service-400-errors"
+  pattern        = "{ $.status = 400 }"
+  log_group_name = aws_cloudwatch_log_group.lpa_uid.name
+
+  metric_transformation {
+    name      = "${local.environment_name}-uid-service-400-errors"
+    namespace = "UID-Service/Monitoring"
+    value     = "1"
+  }
+}
+
+resource "aws_cloudwatch_log_metric_filter" "uid_service_401_errors" {
+  name           = "${local.environment_name}-uid-service-401-errors"
+  pattern        = "{ $.status = 401 }"
+  log_group_name = aws_cloudwatch_log_group.lpa_uid.name
+
+  metric_transformation {
+    name      = "${local.environment_name}-uid-service-401-errors"
+    namespace = "UID-Service/Monitoring"
+    value     = "1"
+  }
+}
+
+resource "aws_cloudwatch_log_metric_filter" "invalid_dob_errors" {
+  name           = "${local.environment_name}-invalid-dob-errors"
+  pattern        = "{$.message = \"*/donor/dob*must match format YYYY-MM-DD*\"}"
```

Comment:
I don't think this will work, since there isn't a `message` property. Instead, you probably need something like:
```
{ $.problem.errors[0].source = "/donor/dob" & $.problem.errors[0].detail = "must match format YYYY-MM-DD" }
```
Unfortunately you can't do something like `errors[*]`, so it doesn't count if there are multiple array items, which might be an acceptable trade-off. An alternative might be joining them all together into a single string for the log.

Reply:
You could also just have a pattern of `"/donor/dob"` to find that string in the entire log entry, which wouldn't tell us what the specific error was but would at least indicate how often each field has a problem.

Reply:
Argh thats annoying that we can't do `errors[*]`. They have a `NOT EXISTS` metric filter but `EXISTS` isn't currently supported which is also a pain. I'll look into joining the strings together